### PR TITLE
Add documentation guide, TOC fixes.

### PR DESCRIPTION
**Pull request by arthurevans**
Also added  icons for the TOC links that open new tabs. 

@robdodson @ebidel @kaycebasques PTAL.

### CODE REVIEW

File: 1.0/tools/documentation.md [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+for information on publishing an element on GitHub, and hosting its API docs
+using GitHub pages.
+{: .alert .alert-info }
+
+
+## View element docs
+
+To view the element docs:
+
+1.  Run `polyserve`:
```
Comment (ebidel) [inline 1.0/tools/documentation.md]: like to tools page on polyserve?

File: 1.0/tools/documentation.md [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+     *
+     *     <body>
+     *       <awesome-sauce></awesome-sauce>
+     *
+     * Wham! It's all awesome now!
+     */
+    Polymer({
+      is: 'awesome-sauce',
+
+You can use Markdown headings to break up a long element summaries:
```
Comment (ebidel) [inline 1.0/tools/documentation.md]: a long element -> long element

File: 1.0/tools/documentation.md [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+*   `@hero`. Specifies a hero image.
+
+        @hero path/to/image
+
+*   `@demo`. Specifies a demo, with optional path and description. If path and description are omitted,
+    the standard demo path (`./demo/`) is assumed.
+
+        @demo
+
+        @demo path/to/demo1  Super cool demo, with sharks!
```
Comment (ebidel) [inline 1.0/tools/documentation.md]: demo1.html and demo2.html

**Comment by ebidel** [comment]
LGTM. Minor nits

**Comment by arthurevans** [comment]
Fixed nits. (Also moved the note about hosting your API docs into the "view your API docs" section. On re-reading, this seemed to make more sense.)

**Comment by addyosmani** [comment]
LGTM 2.

**Pull request merged**